xlators/protocol/client/src/client-rpc-fops_v2.c

```diff
@@ -2196,6 +2197,26 @@ client4_0_lk_cbk(struct rpc_req *req, struct iovec *iov, int count,
         goto out;
     }
 
+    xdata = dict_new();
+    if (xdata == NULL) {
+        rsp.op_errno = ENOMEM;
+        rsp.op_ret = -1;
+        goto out;
+    }
+
+    if (local->cmd == F_GETLK) {
+        conf = this->private;
+        if (!conf)
+            goto out;
+
+        if (conf->strict_locks) {
+            ret = dict_set_int32(xdata, "strict-locks", 1);
+            if (ret)
+                gf_msg_debug(THIS->name, ENOMEM,
+                             "Failed to set strict-locks status into dict");
+        }
+    }
+
     if (rsp.op_ret >= 0) {
         ret = client_post_lk_v2(this, &rsp, &lock, &xdata);
```

Comment:
xdata is getting over-written here. I think you should set this maybe at line 2237

Reply:
Done.